Lib/asyncio/tasks.py

```diff
@@ -496,6 +496,11 @@ async def wait_for(fut, timeout, *, loop=None):
             # after wait_for() returns.
             # See https://bugs.python.org/issue32751
             await _cancel_and_wait(fut, loop=loop)
+            # In case task cancellation failed with some
+            # exception, we should re-raise it
```

Comment:
What about the case when task cancellation failed with some **result** (not exception)? Is it OK to ignore it? I mean something like
```python3
async def inner():
    try:
        await asyncio.sleep(0.2)
    except asyncio.CancellationError:
        return 123  # instead of `raise FooException`
```